### PR TITLE
remote indexes

### PR DESCRIPTION
**Pull request by PRamoneda**
As we speak on Monday. What do you think?

A REMOTES_INDEX dictionary could be added to each loader that needs a remote index.

### CODE REVIEW

File: mirdata/utils.py [hunk 1]
```diff
@@ -233,9 +234,16 @@ def __init__(self, index_file, metadata_load_fn=None):
         self._metadata = None
         self.index_file = index_file
         self.metadata_load_fn = metadata_load_fn
+        self.download_path = download_path
 
     @cached_property
     def index(self):
+        if self.download_path is not None:
```
Comment (nkundiushuti) [inline mirdata/utils.py:241]: this should be tested a bit differently: if the index does not exist locally and if self.download_path is not None

**Comment by nkundiushuti** [comment]
so we finally store big indexes in separate zenodo repositories, right?

**Comment by nkundiushuti** [comment]
what is the impact on the tests for the datasets with remote indexes? 
which tests are affected by this change? 
to check: we will probably would not want to download 2GB when we run a test

File: mirdata/utils.py [hunk 2]
```diff
@@ -216,7 +217,7 @@ def __get__(self, obj, cls):
 
 
 class LargeData(object):
-    def __init__(self, index_file, metadata_load_fn=None):
+    def __init__(self, index_file, metadata_load_fn=None, download_path=None):
```
Comment (magdalenafuentes) [inline mirdata/utils.py:220]: I guess download_path will be used only in the case of loaders with remote indexes?

File: mirdata/utils.py (hunk 1 above)
Comment (magdalenafuentes) [reply mirdata/utils.py:241]: +1

File: mirdata/utils.py (hunk 2 above)
Comment (magdalenafuentes) [reply mirdata/utils.py:220]: I'm a bit confused about what is that you're passing to the downloader in `remotes` below. It should be a dictionary with a `RemoteFileMetadata` object but you're passing a path instead? Or I am missing something?

File: mirdata/utils.py (hunk 1 above)
Comment (PRamoneda) [reply mirdata/utils.py:241]: Yes!! Tomorrow I wil do the tests!

**Comment by PRamoneda** [comment]
> so we finally store big indexes in separate zenodo repositories, right?

Yes!!

**Comment by PRamoneda** [comment]
> what is the impact on the tests for the datasets with remote indexes?
> which tests are affected by this change?
> to check: we will probably would not want to download 2GB when we run a test

mmm ok I have to mangae it then :S

File: mirdata/utils.py (hunk 2 above)
Comment (PRamoneda) [reply mirdata/utils.py:220]: >>I guess download_path will be used only in the case of loaders with remote indexes?
yes, of course!! By default it is not used....

>>I'm a bit confused about what is that you're passing to the downloader in remotes below. It should be a dictionary with a RemoteFileMetadata object but you're passing a path instead? Or I am missing something?
It is a dictionary with a RemoteFileMetadata object or None

Comment (magdalenafuentes) [reply mirdata/utils.py:220]: gotcha, maybe we can call it `remote_index` or something like that then? using the "_path" I'd expect a `str`

**Comment by PRamoneda** [comment]
> what is the impact on the tests for the datasets with remote indexes?

no impact

> which tests are affected by this change?

none

> to check: we will probably would not want to download 2GB when we run a test

It will be only downloaded if the files have not been downloaded before.

File: mirdata/utils.py (hunk 2 above)
Comment (PRamoneda) [reply mirdata/utils.py:220]: done!!!

**Comment by PRamoneda** [comment]
HI!!! It is finished. Codecov errors are not mine.

**Review by nkundiushuti — APPROVED** [review]

**Pull request closed without merging**